diff --git a/node_modules/@bhmb/server/build/extension.js b/node_modules/@bhmb/server/build/extension.js
index 5335ce2..d6b695d 100755
--- a/node_modules/@bhmb/server/build/extension.js
+++ b/node_modules/@bhmb/server/build/extension.js
@@ -7,7 +7,7 @@ var fs = require('fs');
 const express = require('express');
 const serverPort = 8008;
 const serverAddress = 'localhost';
-const jsonParser = bodyParser.json();
+const jsonParser = bodyParser.json({ limit: '10mb' });
 const escapeHtml = (html) => html.replace(/[&<>'"`]/g, ([match]) => ({
     '&': '&amp;',
     '<': '&lt;',
@@ -74,7 +74,7 @@ bot.MessageBot.registerExtension('@bhmb/server', async (ex) => {
             message
         });
         while (messageQueue.length > 500)
-            messageQueue.unshift();
+            messageQueue.shift();
     };
     ex.world.onMessage.sub(({ player, message }) => {
         addMessage(`${player.name}: ${message}`);
